cachito/workers/pkg_managers/yarn.py

```diff
@@ -55,3 +64,293 @@ def get_yarn_proxy_repo_username(request_id):
+
+
+def _convert_to_nexus_hosted(dep_name, dep_source, dep_info):
+    """
+    Convert the input dependency not from the NPM registry to a Nexus hosted dependency.
+
+    :param str dep_name: the name of the dependency
+    :param str dep_source: the source (url or relative path) of the dependency
+    :param dict dep_info: the dependency info from the yarn lock file
+    :return: the dependency information of the Nexus hosted version to use in the yarn lock file
+        instead of the original
+    :raise CachitoError: if the dependency is from an unsupported location or has an unexpected
+        format in the lock file
+    """
+    integrity = dep_info.get("integrity")
+    # TODO: integrity may be a space-separated list of integrity values,
+    #   we ignore the possibility for npm, safe to ignore for yarn as well?
+
+    if not integrity:
+        # For http(s) non-registry dependencies, yarn does not seem to include the "integrity" key
+        # by default. It does, however, include a sha1 hash in the resolved url fragment.
+        url = urlparse(dep_source)
+        if url.fragment and url.scheme in ("http", "https"):
+            integrity = convert_hex_sha_to_npm(url.fragment, "sha1")
+
+    dep = JSDependency(name=dep_name, source=dep_source, integrity=integrity)
+    dep_in_nexus = process_non_registry_dependency(dep)
+
+    converted_dep_info = copy.deepcopy(dep_info)
+    converted_dep_info.update(
+        {
+            "integrity": dep_in_nexus.integrity,
+            "resolved": dep_in_nexus.source,
+            "version": dep_in_nexus.version,
+        }
+    )
+    return converted_dep_info
+
+
+def _get_package_and_deps(package_json_path, yarn_lock_path):
+    """
+    Get the main package and dependencies based on the lock file.
+
+    If the lockfile contains non-registry dependencies, the lock file will be modified to use ones
+    in Nexus. Non-registry dependencies will have the "version_in_nexus" key set.
+
+    :param (str | Path) package_json_path: the path to the package.json file
+    :param (str | Path) yarn_lock_path: the path to the lock file
+    :return: a dictionary that has the keys "deps" which is the list of dependencies,
+        "lock_file" which is the lock file if it was modified (as a dict!), "package" which is the
+        dictionary describing the main package, and "package.json" which is the package.json file if
+        it was modified.
+    :rtype: dict
+    """
+    with open(package_json_path) as f:
+        package_json = json.load(f)
+
+    yarn_lock = pyarn.lockfile.Lockfile.from_file(str(yarn_lock_path)).data
+
+    try:
+        package = {
+            "name": package_json["name"],
+            "version": package_json["version"],
+            "type": "yarn",
+        }
+    except KeyError:
+        raise CachitoError("The package.json file is missing required data (name, version)")
+
+    # TODO: use a different allowlist for yarn? rename allowlist variable?
```

Comment:
Good catch. @lcarva do we want to add support for this here now?

If so, let's add a new allowlist for this one.

Reply:
Added allowlist, we can just keep it empty for now